### DISCUSSION TITLE
how to add a template to a Generic IOC?

### DISCUSSION THREAD

**Discussion by ulrikpedersen** [epics-containers/epics-containers.github.io#122] [Q&A] [answered]
Hi @gilesknap 
Over in my Generic IOC [ioc-ads](https://github.com/ulrikpedersen/ioc-ads), I want to add a DB template but I'm wondering if that is even possible without creating a support module and a ibek-support entry, etc.?
(I also want to add a db substitutions file - but I can see in the tutorials how to do that)

**Comment by gilesknap** [comment]
Hi Ulrik,

Is this template unique to your IOC instance?

If so and it is reasonably small then it can just go in the instance's config folder along side ioc.yaml.

The same is true of a subst file.

One of the tutorials demonstrates adding a little bit of DB into config. See https://epics-containers.github.io/main/tutorials/ioc_changes1.html

**Reply by gilesknap** [reply, marked as answer]
If for some reason you wanted to put a template in the generic IOC but not put it in the support module (maybe this template is in some way specific to containerised IOCs but not traditional ones?) then you could just put it in ioc-ads and copy it into the container with a Dockerfile COPY.

**Reply by gilesknap** [reply]
Maybe I have not understood your requirement in which case give me a call.

**Reply by ulrikpedersen** [reply]
>If so and it is reasonably small then it can just go in the instance's config folder along side ioc.yaml.

OK, I'll try that. Can I then reference the DB template by its filename (so relative path) from the `ioc.subst` file too?

The use-case is an IOC with the adsDriver. This driver is a generic asyn driver that builds its asyn parameter list based on what it finds in the DB. So you have to create a custom db for the application/IOC. I could do it in a support module but it seems a big overhead for just needing a small DB template with 5 records and nothing else...

**Reply by gilesknap** [reply]
It might be worth us having a quick zoom re this. You might as well use absolute path to the DB. You are in a container and you can put it where you like. My first thought is if you are making the DB why not make it fully expanded and not bother with the subst?

If you do need a subst and it is for this one DB only then you can use 'ioc.subst' and the default start.sh will expand that for you. But if the subst needs to expand other templates from the support module too then I don't have a way of supporting that at present other than providing your own start.sh in config too (which could call msi to expand the subst then call ../../start.sh)

**Reply by gilesknap** [reply]
- or in fact /epics/ioc/start.sh to be in keeping with my first point!!!
